src/toil_scripts/adam_pipeline/spark_toil_script.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env python2.7
+"""
+UCSC Computational Genomics Lab RNA-seq Pipeline
+Author: Audrey Musselman-Brown
+Affiliation: UC Santa Cruz Genomics Institute
+
+Please see the README.md in the same directory
+
+Toil pipeline for ADAM preprocessing
+
+                        Tree structure of ADAM pipeline
+                                       0 
+                                       |++(1)
+                                       2 --> 4 --> 5 --> 6 -> 7
+                                        ++(3)
+
+0 = Start Master
+1 = Master Service
+2 = Start Workers
+3 = Worker Service
+4 = Download Data
+5 = ADAM Convert
+6 = ADAM Transform
+7 = Upload Data
+
+================================================================================
+:Dependencies
+docker          - apt-get install docker (or 'docker.io' for linux)
+toil            - pip install --pre toil
+"""
+
+import argparse
+import os
+from subprocess import * #call, check_output
+import sys
+from toil.job import Job
+
+SPARK_MASTER_PORT = "7077"
+HDFS_MASTER_PORT = "8020"
+log = open("python.log", 'a')
+
+# - The tachyon and spark containers need the following ports open:
+#   spark master: 7077, 8080, 4040
+#   tachyon master: 19998, 19999
+#   tachyon worker: 29998, 29999
+
+# JOB FUNCTIONS
+
+
+def start_master(job, inputs):
+    """
+    Starts the master service.
+    """
+    log.write("master job\n")
+    log.flush()
+    masterIP = job.addService(MasterService())
+    job.addChildJobFn(start_workers, masterIP, inputs)
+
+
+def start_workers(job, masterIP,  inputs):
+    """
+    Starts the worker services.
+    """
+    log.write("workers job\n")
+    log.flush()
+    for i in range(inputs['numWorkers']):
+        job.addService(WorkerService(masterIP))
+    job.addFollowOnJobFn(download_data, masterIP, inputs)
+
+
+def call_conductor(masterIP, inputs, src, dst):
+    """
+    Invokes the conductor container.
+    """
+    return call(["docker",
```

Comment:
Use `check_call()` and drop the assertions after every invocation of `call_conductor()`